src/asl.ts

```diff
@@ -2754,13 +2756,32 @@ export class ASL {
       !isReturnStmt(stmt) ||
       predicate.body.statements.length !== 1
     ) {
-      throw new Error(
+      throw new SynthError(
+        ErrorCodes.StepFunction_invalid_filter_syntax,
         'a JSONPath filter expression only supports a single, in-line statement, e.g. .filter(a => a == "hello" || a === "world")'
       );
     }
 
     const toFilterCondition = (expr: Expr): string => {
-      if (isBinaryExpr(expr)) {
+      const constant = evalToConstant(expr);
+      if (constant) {
+        if (typeof constant.constant === "string") {
+          return `'${constant.constant.replace(/'/g, "\\'")}'`;
+        }
```

Comment:
comment on why this magic is needed

Reply:
I am not sure, actually, I copies it from the `isStringLiteralExpr` I removed (because evalToConstant covers the case).